workflows/taxprofiler.nf

```diff
@@ -108,43 +91,106 @@ include { CAT_FASTQ as MERGE_RUNS     } from '../modules/nf-core/cat/fastq/main'
+    samplesheet
+        .branch { meta, run_accession, instrument_platform, fastq_1, fastq_2, fasta ->
+            //println "Mapping: meta=$meta, run_accession=$run_accession, instrument_platform=$instrument_platform, fastq_1=$fastq_1, fastq_2=$fastq_2, fasta=$fasta"
 
+        meta.run_accession = run_accession
+        meta.instrument_platform = instrument_platform
 
-    DB_CHECK (
-        ch_databases
-    )
-    ch_versions = ch_versions.mix(DB_CHECK.out.versions)
+        // Define single_end based on the conditions
+        meta.single_end = (fastq_1 && !fastq_2 && instrument_platform != 'OXFORD_NANOPORE')
+
+        // Define is_fasta based on the presence of fasta
+        meta.is_fasta = fasta ? true : false
+
+        if (!meta.is_fasta && !fastq_1) {
+            error("ERROR: Please check input samplesheet: entry `fastq_1` doesn't exist!")
+        }
+        if (meta.instrument_platform == 'OXFORD_NANOPORE' && fastq_2) {
+            error("Error: Please check input samplesheet: for Oxford Nanopore reads entry `fastq_2` should be empty!")
+        }
+        if (meta.single_end && fastq_2) {
+            error("Error: Please check input samplesheet: for single-end reads entry `fastq_2` should be empty")
+        }
+        // create fastq_se channel if single_end
+        fastq_se: meta.single_end
+            return [meta, [fastq_1]]
+        //
+        nanopore: instrument_platform == 'OXFORD_NANOPORE'
+            return [meta, [fastq_1]]
+        fastq_pe: fastq_2
+            return [meta, [fastq_1, fastq_2]]
+        ch_fasta: meta.is_fasta && meta.single_end
+            return [meta, [fasta]]
+    }
+    .set { ch_input }
+
+    // Merge ch_input.fastq_pe and ch_input.fastq_se into a single channel
+    def ch_fastq = ch_input.fastq_pe.mix(ch_input.fastq_se)
+    // Merge ch_fastq and ch_input.nanopore into a single channel
+    def ch_input_for_fastqc = ch_fastq.mix(ch_input.nanopore)
+
+    // Validate databases
+    databases
+        .map { db_meta, db -> [db_meta.db_params]
+            def corrected_db_params = db_meta.db_params == null ? '' : db_meta.db_params
+            db_meta.db_params = corrected_db_params
+            // Check the empty db_params is ""
+            println "db_params: $db_meta.db_params"
+        }
+        .set { ch_databases }
+
+    // Equivenent to "uniqueEntries" in schema_databases.json, but with better error message. Should we remove this part?
+    // ch_databases
+    //     .map {db_meta, db -> [db_meta.tool, db_meta.db_name] }
+    //     .groupTuple()
+    //     .map { tool, db_name ->
+    //         def unique_names = db_name.unique(false)
+    //             if (unique_names.size() < db_name.size()) {
+    //                 error("[nf-core/taxprofiler] ERROR: Each database for a tool must have a unique name, duplicates detected. Tool: ${tool}, Database names: ${unique_names}")
+    //             }
+    //     }
```

Comment:
do you still want this in the code?

Reply:
I prefer the error message from this code over the one from `uniqueEntries`. But, using this code is more complicated compared to using `uniqueEntries`. What would be your suggestion?

Reply:
@sofstam What do you think? Should we removed it?

Reply:
I was more thinking if this commented out code was on oversight from some or if it is supposed to stay in.